src/Razor/test/Microsoft.VisualStudio.LanguageServerClient.Razor.Test/HtmlCSharp/GoToImplementationHandlerTest.cs

```diff
@@ -0,0 +1,171 @@
+            var invokedRemapRequest = false;
+            var expectedLocation = GetLocation(5, 5, 5, 5, Uri);
+            var documentManager = new TestDocumentManager();
+            documentManager.AddDocument(Uri, Mock.Of<LSPDocumentSnapshot>());
+
+            var virtualCSharpUri = new Uri("C:/path/to/file.razor.g.cs");
+            var csharpLocation = GetLocation(100, 100, 100, 100, virtualCSharpUri);
+            var requestInvoker = new Mock<LSPRequestInvoker>(MockBehavior.Strict);
+            requestInvoker
+                .Setup(r => r.ReinvokeRequestOnServerAsync<TextDocumentPositionParams, Location[]>(It.IsAny<string>(), It.IsAny<LanguageServerKind>(), It.IsAny<TextDocumentPositionParams>(), It.IsAny<CancellationToken>()))
+                .Callback<string, LanguageServerKind, TextDocumentPositionParams, CancellationToken>((method, serverKind, implementationParams, ct) =>
+                {
+                    Assert.Equal(Methods.TextDocumentImplementationName, method);
+                    Assert.Equal(LanguageServerKind.CSharp, serverKind);
+                    invokedLSPRequest = true;
+                })
+                .Returns(Task.FromResult(new[] { csharpLocation }));
+
+            var projectionResult = new ProjectionResult()
+            {
+                LanguageKind = RazorLanguageKind.CSharp,
+            };
+            var projectionProvider = new Mock<LSPProjectionProvider>(MockBehavior.Strict);
+            projectionProvider.Setup(p => p.GetProjectionAsync(It.IsAny<LSPDocumentSnapshot>(), It.IsAny<Position>(), It.IsAny<CancellationToken>())).Returns(Task.FromResult(projectionResult));
+
+            var documentMappingProvider = new Mock<LSPDocumentMappingProvider>();
+            documentMappingProvider
+                .Setup(d => d.RemapLocationsAsync(It.IsAny<Location[]>(), It.IsAny<CancellationToken>()))
+                .Callback<Location[], CancellationToken>((locations, token) =>
+                {
+                    Assert.Equal(csharpLocation, locations[0]);
+                    invokedRemapRequest = true;
+                })
+                .Returns(Task.FromResult(Array.Empty<Location>()));
+
+            var implementationHandler = new GoToImplementationHandler(requestInvoker.Object, documentManager, projectionProvider.Object, documentMappingProvider.Object);
+            var implementationRequest = new TextDocumentPositionParams()
+            {
+                TextDocument = new TextDocumentIdentifier() { Uri = Uri },
+                Position = new Position(10, 5)
+            };
+
+            // Act
+            var result = await implementationHandler.HandleRequestAsync(implementationRequest, new ClientCapabilities(), CancellationToken.None).ConfigureAwait(false);
+
+            // Assert
+            Assert.True(invokedLSPRequest);
+            Assert.True(invokedRemapRequest);
+
+            // Actual remapping behavior is tested elsewhere.
+        }
+
```

Comment:
Is a null results check needed or would it be overkill?

https://github.com/dotnet/aspnetcore-tooling/pull/1967/files#diff-1c8e30f00c504daba13aeccac7de6fbbR186-R216

Reply:
I wouldn't call it overkill. Imo it is ideal to have it but we generally aren't exhaustive to that level. There are many other places we don't test to that extent unless it is a hot path (e.g. the parser) or it is part of an interesting complex scenario. I wouldn't remove it from your PR though.